soc/realtek/ameba/amebad/soc.c

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2024 Realtek Semiconductor Corp.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <soc.h>
+#include <ameba_soc.h>
+#include <zephyr/device.h>
+#include <zephyr/init.h>
+#include <zephyr/kernel.h>
+#include <zephyr/cache.h>
+#include "ameba_system.h"
+
+void z_arm_reset(void);
+
+IMAGE2_ENTRY_SECTION
+RAM_START_FUNCTION Img2EntryFun0 = {z_arm_reset, NULL, /* BOOT_RAM_WakeFromPG, */
+				    (uint32_t)NewVectorTable};
+
+static void app_vdd1833_detect(void)
+{
+	u32 temp;
+
+	if (FALSE == is_power_supply18()) {
```

Comment:
could you use `false` and `stdbool.h`? also, `if (is_power_supply18() == false) {` or `if (!is_power_supply18()) {`, the thing about putting the constant before the `==` is weird, I know some people like it but it's kindof avoided in Zephyr